### PR TITLE
Final major preparations

### PR DESCRIPTION
**Pull request by core23**
## Subject

I am targeting this branch, because this is (hopefully) the last PR to get a new stable release.

Closes #638

## Changelog

```markdown
### Added
- Add `BlockLoaderInterface::exists` method to interface
- Add `BlockContextManagerInterface::exists` method to interface

### Changed
- Make all services public by default

### Removed
- Remove service class parameters
- Remove PageBundle specific code
- Remove addClassesToCompile

### Fixed
- Fix twig import of macros
- Load menu block only when KnpMenuBundle exist
- Remove undefined method call
```

### CODE REVIEW

**Comment by core23** [comment]
Blocking by https://github.com/sonata-project/dev-kit/pull/513

**Comment by core23** [comment]
Please review @sonata-project/contributors

File: src/Templating/Helper/BlockHelper.php [hunk 1]
```diff
@@ -325,13 +322,14 @@ private function getEventBlocks(BlockEvent $event): array
...
     }
 
-    /**
-     * @internal since sonata-project/block-bundle 3.16
-     */
     private function getEventListeners(string $eventName): array
     {
         $results = [];
 
+        if (!$this->eventDispatcher instanceof \Symfony\Component\EventDispatcher\EventDispatcherInterface) {
```
Comment (phansys) [inline src/Templating/Helper/BlockHelper.php]: Could we import this class?

Comment (core23) [reply src/Templating/Helper/BlockHelper.php]: We could use an alias, because we have an existing interface with the same name

**Pull request merged**